src/node_file.cc

```diff
@@ -201,8 +298,19 @@ static void After(uv_fs_t *req) {
         break;
 
       case UV_FS_READLINK:
-        argv[1] = String::NewFromUtf8(env->isolate(),
-                                      static_cast<const char*>(req->ptr));
+        link = EncodeFilename(env,
+                              static_cast<const char*>(req->ptr),
+                              req_wrap->encoding_);
+        if (link->IsNull()) {
```

Comment:
When does EncodeFilename() return `v8::Null()`?  It returns an empty handle on error, doesn't it?